### PR TITLE
Refactor event payload to data

### PR DESCRIPTION
**Pull request by kodumbeats**
## What does this PR do?

This PR refactors the function environment variable `APPWRITE_FUNCTION_EVENT_PAYLOAD` to `APPWRITE_FUNCTION_EVENT_DATA`. Includes breaking change, so keeping on top of `0.8.x` for now.

## Test Plan

Existing test coverage on webhooks

## Related PRs and Issue

Harmonization required after #967 

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
 - Only logged in users can execute functions (for guests, use anonymous login)
 - Only the user who has triggered the execution get access to the relevant execution logs
+- Function execution env `APPWRITE_FUNCTION_EVENT_PAYLOAD` renamed to `APPWRITE_FUNCTION_ENV_DATA`
```
Comment (eldadfux) [inline CHANGES.md:17]: to `APPWRITE_FUNCTION_EVENT_DATA`

Comment (eldadfux) [reply CHANGES.md:17]: I would also list this change under breaking changes.

File: app/workers/functions.php [hunk 2]
```diff
@@ -147,7 +147,7 @@ public function perform()
         $trigger = $this->args['trigger'] ?? '';
         $event = $this->args['event'] ?? '';
         $scheduleOriginal = $this->args['scheduleOriginal'] ?? '';
-        $payload = (!empty($this->args['payload'])) ? json_encode($this->args['payload']) : '';
+        $eventData= (!empty($this->args['eventData'])) ? json_encode($this->args['eventData']) : '';
```
Comment (eldadfux) [inline app/workers/functions.php:150]: missing space before `=`

**Review by eldadfux — CHANGES_REQUESTED** [review]
Looks good, minor comments.

File: CHANGES.md (hunk 1 above)
Comment (kodumbeats) [reply CHANGES.md]: This line currently under `0.8 Breaking Changes`. Is there another place to list breaking changes?
https://github.com/appwrite/appwrite/blob/456793ebd8dc92e67b5643d2e95775a5da88f6e8/CHANGES.md#breaking-changes

Comment (eldadfux) [reply CHANGES.md]: Sorry my bad, had the wrong context view on the file changes

**Review by eldadfux — APPROVED** [review]

**Comment by eldadfux** [comment]
Waiting for tests

**Comment by eldadfux** [comment]
🚀

**Pull request merged**